### ISSUE TITLE
Document that a Specialize'd scheduler module passes the rig's purity check

### ISSUE THREAD

**Issue by Eric-Louchez** [Shorokoo/Shorokoo#216]
`training.md` requires a scheduler module's inputs to be "a named subset of the reserved int64 scalar
counters {step, epoch, batchIndex}", and says the rig rejects a scheduler graph carrying "an
unrecognized input".

That leaves an obvious pattern undefined: declaring the schedule's milestones as `[Hyper]` and
stripping them with `Specialize` before handing the graph to `Hyperparameter.Scheduled`, so the
milestones stay run-configurable without becoming rig inputs.

```csharp
[Module] public partial class Sched
{
    public static Scalar<float32> Inline(Scalar<int64> step, [Hyper(10)] Scalar<int32> total)
        => Scalar(0.1f) * (Scalar(1f) - step.Cast<float32>() / total.Cast<float32>());
}

var g    = Sched.ComputationGraph;
var spec = g.Specialize(g.FromOrderedInputs([TensorData([], new[] { 10 })]));
var rig  = TrainingRig.FromScratch(model, loss, AdamOptimizer.ComputationGraph, sampleInputs,
    new AdamOptimizerHyperparameters { LearningRate = Hyperparameter.Scheduled(spec) });
```

I have now verified this **is** accepted, and the schedule drives the optimizer correctly:

```
step 1: |delta| = 0.100000
step 2: |delta| = 0.090000
step 3: |delta| = 0.080000
```

Nothing in the docs said so, and the cost of finding out experimentally is a full rig build — which
for a real model is minutes. Worth one sentence on the `Hyperparameter.Scheduled` documentation
stating that a `Specialize`d module graph satisfies the purity check, since it is the natural way to
keep schedule milestones configurable.
